### PR TITLE
Replace recursive QMutexes with QRecursiveMutex

### PR DESCRIPTION
**Pull request by Holzhaus**
Recursive QMutexes have been deprecated since the 5.15.

Fixes warnings like these:

    src/vinylcontrol/vinylcontrolprocessor.cpp: In constructor ‘VinylControlProcessor::VinylControlProcessor(QObject*, UserSettingsPointer)’:
    src/vinylcontrol/vinylcontrolprocessor.cpp:28:32: error: ‘QMutex::QMutex(QMutex::RecursionMode)’ is deprecated: Use QRecursiveMutex instead of a recursive QMutex [-Werror=deprecated-declarations]
       28 |           m_bReloadConfig(false) {
          |                                ^
    In file included from /usr/include/qt/QtCore/QMutex:1,
                     from src/vinylcontrol/vinylcontrolprocessor.h:6,
                     from src/vinylcontrol/vinylcontrolprocessor.cpp:1:
    /usr/include/qt/QtCore/qmutex.h:140:14: note: declared here
      140 |     explicit QMutex(RecursionMode mode);
          |              ^~~~~~

### CODE REVIEW

File: src/engine/controls/cuecontrol.cpp [hunk 1]
```diff
@@ -85,7 +85,11 @@ CueControl::CueControl(const QString& group,
           m_bypassCueSetByPlay(false),
           m_iNumHotCues(NUM_HOT_CUES),
           m_pLoadedTrack(),
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+          m_mutex() {
```
Comment (uklotzde) [inline src/engine/controls/cuecontrol.cpp:89]: No need to explicitly invoke the default constructor. Also applies to `m_pLoadedTrack()` above.

```
#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
{
#else
, m_mutex(QMutex::Recursive) {
#endif
```

File: src/util/mutex.h [hunk 2]
```diff
@@ -12,8 +12,8 @@
 
 class CAPABILITY("mutex") MMutex {
   public:
-    MMutex(QMutex::RecursionMode mode = QMutex::NonRecursive)
-            : m_mutex(mode) {
+    MMutex()
```
Comment (uklotzde) [inline src/util/mutex.h:15]: = default;

File: src/vinylcontrol/vinylcontrolprocessor.cpp [hunk 3]
```diff
@@ -20,7 +20,11 @@ VinylControlProcessor::VinylControlProcessor(QObject* pParent, UserSettingsPoint
           m_pConfig(pConfig),
           m_pToggle(new ControlPushButton(ConfigKey(VINYL_PREF_KEY, "Toggle"))),
           m_pWorkBuffer(SampleUtil::alloc(MAX_BUFFER_LEN)),
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
```
Comment (uklotzde) [inline src/vinylcontrol/vinylcontrolprocessor.cpp:23]: Only `QT_VERSION < QT_VERSION_CHECK(5, 14, 0)` is required, no else.

**Review by uklotzde — CHANGES_REQUESTED** [review]

**Comment by Holzhaus** [comment]
Merge? Currently I need to pull it into every branch I test to be able to build with `-DWARNINGS_FATAL=ON`.

**Review by uklotzde — APPROVED** [review]
Sorry for the delay, busy times. LGTM

**Pull request merged**